src/unix/uiactionx11.cpp

```diff
@@ -46,6 +55,113 @@ GtkWidget* wxGetTopLevelGTK();
 
 namespace
 {
+// Like the real events, this class tries to add a _fake_ delay to the generated
+// (fake) events so that chances that they got lost or ignored (for whatever reasons)
+// by the X server (or the WM, or even the input device driver) are minimized.
+class wxXSync
+{
+public:
+    wxXSync(wxX11Display& display)
+        : m_display(display), m_isMotion(true)
+    {
+    }
+
+    wxXSync(wxX11Display& display, bool depressed)
+        : m_display(display), m_isMotion(false)
+    {
+        static int s_maxDepressed = 0;
```

Comment:
I don't understand why is this needed, is it just for debugging wrong use of this class (and what does wrong actually mean here anyhow? It would be nice to document this in the class comment above...)?

Reply:
`s_maxDepressed` helps to :

- detect erroneous calls to `KeyUp/MouseUp` before a prior calls to `KeyDown/MouseDown`.
- make sure all keys are down before trying to release any key.
- sleep only once: when all keys are down and after the last depressed key is up.

For ex.: 
`sim.Char('F', wxMOD_ALT);
`treated as
```
KeyDown('F');
KeyDown(wxMOD_ALT);
wxYield();
wxMilliSleep(Default_Delay);
KeyUp(wxMOD_ALT);
KeyUp('F');
wxYield();
wxMilliSleep(Default_Delay);
```